doc/locale_api.md

```diff
@@ -0,0 +1,283 @@
+byte* tagging the priority. For ease of recognition when watching bus traffic,
+special numbers are used:
+- `23` means Human, for the number of chromosome pairs
+- `42` means Machine, as the famous Answer was given by Deep Thought, a machine
+
+In the following dump, we see that the locale was set by the user and the
+system has adjusted the keyboard.
+
+```
+node ...Agama/Locale1 {
+  interface ...Agama.Locale1 {
+    properties:
+      readwrite (yas)   Locale = (23, ['cs_CZ.UTF-8', 'de_DE.UTF-8']);
+      readwrite (y(ss)) X11Keyboard = (42, ('cz','qwerty));
+  };
+};
+```
+
+You may know a [similar settings in libzypp][resstatus] where it has 4 levels.
+
+*: maybe this is a crazy optimization? I am not too opposed to use strings for
+this on the bus.
+
+[resstatus]: https://github.com/openSUSE/libzypp/blob/d441746c59f063b5d54833bfdebc48829b07feb5/zypp/ResStatus.h#L106
+
+
+## Interfaces
+
+### Language and Keyboard
+
+- when setting the locale, adjust the proposed package selection and keyboard
+  accordingly. And timezone.
+
+The general design of the proposal layer is
+
+- declarative, using read-write properties
+- setting some properties will make changes in the proposal layer of other
+  properties of other objects
+
+I don't know: should the proposal be adjusted automatically as part of the property setter, or should it be explicit?
+
+So here, setting `Locale` below will set also `VConsoleKeyboard` here and
+  - Agama...Software...todo(...)
+  - Agama...Timezone...todo(...)
+
+For the first version of the API, let's keep things simple:
+
+**LocaleType** is just one string, the value for the `LANG` variable, like
+`"cs_CZ.UTF-8"`.
+
+**VConsoleKeyboardType** is a string, for example
+`"cz-qwerty"` or `"us"`.
+
+`systemd-firstboot` only has an option for the console keymap, but we have a
+way to propagate it to X11, see [bsc#1046436](https://bugzilla.suse.com/show_bug.cgi?id=1046436)
+
+We don't expose the X11 keyboard, instead letting systemd do it via the
+_convert_ parameter.
+
+(The other systemd keyboard settings are X11Model and X11Options, we don't
+have UI or data for that)
+
+**FIXME:** _langtable_ on the other hand only deals with the X11 keyboards.
+And legacy YaST has the console keyboard as the primary key. Must resolve this.
+
+```
+# this is gdbus syntax BTW
+node /org/opensuse/Agama/Locale1 {
+  interface org.opensuse.Agama.Locale1 {
+    methods:
+      ListLocales(
+        out a(sss) id_english_native # [('cs_CZ.UTF-8', 'Czech', 'Čeština'),…]
+      )
+      # langtable has X11 keyboards and no localization of their labels
+      ListX11Keyboards(
+        out as    ids  # id like "layout(variant)"
+      )
+
+      # ProposeKeyboard(); # not needed? adjusted automatically, same object
+      # Sets Agama/TimeDate1's Timezone (but not LocalRTC, that's for Storage to say?)
+      ProposeTimeDate(); # different object but same service
+      ProposeSoftware(); # different service
```

Comment:
if we use properties, it should not need method and instead software should observe property for change, not?